packages/dataviews/src/dataform-controls/checkbox.tsx

```diff
@@ -15,11 +19,38 @@ export default function Checkbox< Item >( {
 	hideLabelFromVision,
 }: DataFormControlProps< Item > ) {
 	const { id, getValue, label, description } = field;
+	const [ customValidity, setCustomValidity ] =
+		useState<
+			React.ComponentProps<
+				typeof ValidatedCheckboxControl
+			>[ 'customValidity' ]
+		>( undefined );
 
 	return (
-		<CheckboxControl
-			__nextHasNoMarginBottom
+		<ValidatedCheckboxControl
+			required={ !! field.isValid?.required }
+			onValidate={ ( newValue: any ) => {
+				const message = field.isValid?.custom?.(
+					{
+						...data,
+						[ id ]: newValue,
+					},
+					field
+				);
+
+				if ( message ) {
+					setCustomValidity( {
+						type: 'invalid',
+						message,
+					} );
+					return;
+				}
+
+				setCustomValidity( undefined );
+			} }
+			customValidity={ customValidity }
 			hidden={ hideLabelFromVision }
+			__nextHasNoMarginBottom
```

Comment:
This is already part of the validatedcheckbox control, [right](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/validated-form-controls/components/checkbox-control.tsx#L52)?

```suggestion
```